src/libasr/codegen/wasm_to_x64.cpp

```diff
@@ -309,6 +326,23 @@ class X64Visitor : public WASMDecoder<X64Visitor>,
         handleI32Opt([&](){ m_a.asm_xor_r64_r64(X64Reg::rax, X64Reg::rbx);});
     }
 
+    void visit_I32Store(uint32_t /*mem_align*/, uint32_t /*mem_offset*/) {
+        m_a.asm_pop_r64(X64Reg::rbx);
+        m_a.asm_pop_r64(X64Reg::rax);
+        // Store value rbx at location rax
+        // X64Reg base = X64Reg::rax;
+        // m_a.asm_mov_m64_r64(&base, nullptr, 1, 0, X64Reg::rbx);
+
+        // TODO: Memory values shared by wasm correspond to i32 integers
+        // Storing each integer using 8 bytes could lead
+        // to different memory location from that shared by wasm.
+        // Fix this in upcoming PRs.
+        m_a.asm_mov_r32_label(X86Reg::esi, "master_string"); // base_memory
+        m_a.asm_add_r32_r32(X86Reg::esi, X86Reg::eax); // base_memory + location
+        X86Reg base = X86Reg::esi;
+        m_a.asm_mov_m32_r32(&base, nullptr, 1, 0, X86Reg::ebx);
```

Comment:
**Line 343:** The assembly code segment faults at this line. I am unsure, but it seems the memory area where we are storing strings (and their location, length information) probably is read only. If that is the case, we might need to search for other options.

I will make a small example and confirm if the memory area is actually read-only or not.

Reply:
Yes, it seems it is read-only.

**Small Example:**
```asm
_start:
    mov r10, base_memory
    mov r8, 15
    mov [r10], r8 ; writing to base_memory location
    mov rdi, 23 ; custom exit code
    mov rax, 60
    syscall

base_memory:
    db 0x20
    db 0x20
    db 0x20
    db 0x20
```
```bash
lpython$ nasm -fbin simple.asm && chmod +x simple && ./simple
Segmentation fault (core dumped)
lpython$ echo $?
139
```

Reply:
**PS:** I figured a better approach which could/would tackle it (and hopefully make the strings implementation simpler). It (hopefully) would also fix the inefficiency shared at https://github.com/lcompilers/lpython/pull/1497#discussion_r1097932266.

Reply:
The better approach involves directly encoding the `iov` vector into a data segment (i.e. in the form of a string). This approach works in `wasm` (and hopefully would work with `wasm_x64`) but the `wat` generated doesn't seem to print any output.

**PS:** This was occurring due to incorrect conversion of `byte` to string. Using `sprintf(byte.data(), "%02x", uint8_t(ch))` instead of `sprintf(byte.data(), "%02x", int(ch))` seems to fix the issue.